Add an auto refresh toggle when relative timeframe is set

kennethnym opened the pull request:
## Description
This PR adds a toggle next to the refresh button that allows users to enable auto refresh of data when a relative time frame is set, e.g. "last 10 minutes".

I set the auto refresh to happen every minute, but happy to change it if needed.

## Testing instructions

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage
- [ ] {more steps here}

## Agile board tracking
connect to DSEGOG-86

kennethnym committed:
Add an auto refresh toggle when timeframe set

kennethnym committed:
Fix unit tests

louise-davies reviewed (changes requested):
Code looks good, however when a user first loads up the app, the first time they select a relative timeframe the auto refresh toggle quickly flashes into view then away again. See GIF

![OG-auto-refresh](https://user-images.githubusercontent.com/22525471/233022383-4d4f2432-4d84-4dd5-abcc-eb25c09f3d55.gif)

Could you investigate this please? Also, I believe that the auto-refresh toggle should have the same "behaviour" as the refresh button itself - either both the buttom and the toggle should be hidden when a non-relative timeframe search is used, or both are visible but disabled. It's a bit odd to have one doing one and one doing the other. If we decide to just disable them both then that fies the issue in the GIF, but I wouldn't mind hiding them both if the GIF issue is fixed.

kennethnym commented:
I think I know why it is happening - the toggle's appearance depends on `timeframeRange` & `paramsUpdated`. When the user selects a time frame, `timeframeRange` is set immediately, but `paramsUpdated` at that time is still `false`, so the toggle briefly appears, then `paramsUpdated` is correctly updated in a `useEffect`, which then hides the toggle.

kennethnym committed:
Make auto refresh toggle always visible

Auto refresh toggle is now always visible but disabled when relative timeframe is not set

kennethnym committed:
Update tests for the new auto refresh behavior

louise-davies reviewed (approved):
LGTM

louise-davies reviewed (approved):
LGTM
